lib/src/configuration.cc

```diff
@@ -280,6 +282,41 @@ HW::ParseResult Configuration::parseOptions(int argc, char *argv[])
     return parse_result;
 }
 
+#ifdef HAS_LTH_SYSLOG
+lth_log::syslog_facility string_to_syslog_facility(std::string facility)
+{
+    lth_log::syslog_facility fac;
+    try {
+        // NOTE(ale): ignoring HorseWhisperer's vlevel ("-v" flag)
+        const std::map<std::string, lth_log::syslog_facility> option_to_syslog_facility {
+            { "kern", lth_log::syslog_facility::kern },
+            { "user", lth_log::syslog_facility::user },
+            { "mail", lth_log::syslog_facility::mail },
+            { "daemon", lth_log::syslog_facility::daemon },
+            { "auth", lth_log::syslog_facility::auth },
+            { "syslog", lth_log::syslog_facility::syslog },
+            { "lpr", lth_log::syslog_facility::lpr },
+            { "news", lth_log::syslog_facility::news },
+            { "uucp", lth_log::syslog_facility::uucp },
+            { "cron", lth_log::syslog_facility::cron },
+            { "local0", lth_log::syslog_facility::local0 },
+            { "local1", lth_log::syslog_facility::local1 },
+            { "local2", lth_log::syslog_facility::local2 },
+            { "local3", lth_log::syslog_facility::local3 },
+            { "local4", lth_log::syslog_facility::local4 },
+            { "local5", lth_log::syslog_facility::local5 },
+            { "local6", lth_log::syslog_facility::local6 },
+            { "local7", lth_log::syslog_facility::local7 },
+        };
+        fac = option_to_syslog_facility.at(facility);
+    } catch (const std::out_of_range& e) {
+        throw Configuration::Error {
+            lth_loc::format("invalid syslog facility: '{1}'", facility) };
+    }
+    return fac;
+}
+#endif
+
```

Comment:
I think this may better belong in leatherman, and pass string to `setup_syslog_logging`